brd-android/kyc/src/main/java/com/fabriik/kyc/ui/features/accountverification/AccountVerificationViewModel.kt

```diff
@@ -91,14 +106,28 @@ class AccountVerificationViewModel(
                             )
                     }
                 }
+
+            is AccountVerificationContract.Event.ProfileLoaded ->
+                setState {
+                    AccountVerificationContract.State(
+                        profile = event.profile,
+                        level1State = mapStatusToLevel1State(event.profile.kycStatus),
+                        level2State = mapStatusToLevel2State(event.profile.kycStatus),
+                        isLoading = false
+                    )
+                }
+
+            is AccountVerificationContract.Event.ProfileLoadFailed ->
+                setEffect { AccountVerificationContract.Effect.ShowToast(event.message) }
         }
     }
 
-    private fun mapStatusToLevel1State(status: KycStatus): AccountVerificationContract.Level1State {
+    private fun mapStatusToLevel1State(status: KycStatus?): AccountVerificationContract.Level1State {
         val state = when (status) {
             KycStatus.DEFAULT,
             KycStatus.EMAIL_VERIFIED,
-            KycStatus.EMAIL_VERIFICATION_PENDING -> null
+            KycStatus.EMAIL_VERIFICATION_PENDING,
+            null -> null
```

Comment:
Could this be refactored in a more verbose way?. i.e. state could be something like "AccountVerificationStatusView.StatusViewState.Unverified" instead of "null"

Reply:
Am since we only get the iconRes from this state and we already cover null exception in function that sets the icon (where the default icon is set in null case), I think it's okey to leave it nullable. Maybe this should still be refactored but I don't think it is related to this PR.

Reply:
Consider adding TODO here, so that we don't forget to change it